cmd/chart-repo/utils.go

```diff
@@ -85,14 +87,30 @@ func init() {
 // imported into the database as fast as possible. E.g. we want all icons for
 // charts before fetching readmes for each chart and version pair.
 func syncRepo(dbSession datastore.Session, repoName, repoURL string, authorizationHeader string) error {
-	url, err := parseRepoUrl(repoURL)
+	url, err := parseRepoURL(repoURL)
 	if err != nil {
 		log.WithFields(log.Fields{"url": repoURL}).WithError(err).Error("failed to parse URL")
 		return err
 	}
 
 	r := repo{Name: repoName, URL: url.String(), AuthorizationHeader: authorizationHeader}
-	index, err := fetchRepoIndex(r)
+	repoBytes, err := fetchRepoIndex(r)
+	if err != nil {
+		return err
+	}
+
+	repoChecksum, err := getSha256(repoBytes)
+	if err != nil {
+		return err
+	}
+
+	// Check if the repo has been already processed
+	if repoAlreadyProcessed(dbSession, repoName, repoChecksum) {
+		log.WithFields(log.Fields{"url": repoURL}).Info("Skipping repository since it has been already processed")
```

Comment:
perhaps this should say since there are no new updates instead, processed implies a one-time thing